src/npc.c

```diff
@@ -358,7 +358,45 @@ INCLUDE_ASM(s32, "npc", npc_do_player_collision, Npc* npc);
 
 INCLUDE_ASM(void, "npc", npc_do_gravity, Npc* npc);
 
-INCLUDE_ASM(s32, "npc", func_800397E8);
+s32 func_800397E8(Npc* npc, f32 arg1) {
+    if (!(npc->flags & 0x208))
+    {
+        f32 x;
+        f32 y;
+        f32 z;
+        f32 subroutine_arg;
+        f32 temp_v1;
+        s32 phi_v0;
+
+        if (npc->flags & 0x800) {
+            npc->flags &= ~0x1000;
+            return 0;
+        }
+
+        temp_v1 = fabsf(arg1) + 16;
+        subroutine_arg = temp_v1;
+        x = npc->pos.x;
+        y = npc->pos.y + 13;
+        z = npc->pos.z;
+
+        if (!(npc->flags & 0x4000000)) {
+            phi_v0 = npc_raycast_down_sides(npc->unk_80, &x, &y, &z, &subroutine_arg);
+        } else {
+            phi_v0 = npc_raycast_down_ahead(npc->unk_80, &x, &y, &z, &subroutine_arg, npc->yaw,
+                                            npc->collisionRadius);
+        }
+
+        if (phi_v0 && subroutine_arg <= temp_v1) {
```

Comment:
Because phi_v0 is a number here, rather than a pointer, this should be `phi_v0 != 0`